x-pack/plugins/security_solution/public/resolver/test_utilities/simulator/index.tsx

```diff
@@ -175,14 +175,24 @@ export class Simulator {
   }
 
   /**
-   * Return an Enzyme ReactWrapper for any child elements of a specific processNodeElement
-   *
-   * @param entityID The entity ID of the proocess node to select in
-   * @param selector The selector for the child element of the process node
+   * The button that opens a node's submenu.
    */
-  public processNodeChildElements(entityID: string, selector: string): ReactWrapper {
+  public processNodeSubmenuButton(
+    /** nodeID for the related node */ entityID: string
+  ): ReactWrapper {
     return this.domNodes(
-      `${processNodeElementSelector({ entityID })} [data-test-subj="${selector}"]`
```

Comment:
:grey_question: This is one case where the structure of the selector before ("button that is owned by the process") was valuable I think. I don't see that reflected as much in simplifying it to "a button with an ID". Someone could move that button out from the node, changing the semantics and this test would still pass.

Reply:
How exactly would removing a wrapper div change the semantics?

Reply:
The wrapper gets an `aria-selected` attribute. is that what you're thinking about?

Reply:
Even though the submenu button gets tagged with the `data-test-resolver-node-id` that doesn't really say the same thing as the button being contained by the node. A well-meaning dev could move that button outside the node (while finding another way to populate `data-test-resolver-node-id` as the test expects here and break the semantic meaning without tripping the test up at all. I don't see that as an "improvement" to what's there or easier to read. We could discuss at office hours.

Reply:
I guess we could assert this "contains" stuff in the test itself, but then to Michael's point earlier, it kinda beggars the purpose of this separate method altogether.

Reply:
> A well-meaning dev could move that button outside the node (while finding another way to populate data-test-resolver-node-id as the test expects here
That's exactly what I'm going for. Why would we want that to break the test? Is there some observable behavior that's being broken?

Reply:
:straight_ruler: 

Post discussion at office hours: We _do_ want to test things like that buttons are contained by certain elements (say with certain aria-roles or they have some other kind of equivalent aria- relationship like `owns` ) but that is perhaps better to assert in individual tests (i.e. write a test that asserts that all sumbenu button are children of a `role=treeitem` or have some equivalent structural relationship established via other attributes.)

That allows us to keep methods on the simulator more simple/single-purpose ("just get this element") and write tests for the things we care to assert.

Reply:
we discussed this. The DOM structure is important. These abstractions are still useful. With them, not every tests needs to be coupled to the DOM structure. These methods may change in the future to rely on the DOM structure, that should be fine. Also, we will have separate tests that check the DOM structure and a11y roles.